src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -157,6 +158,11 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
             }
         }
 
+        @Override public byte[] getOutput(FilePath workspace, Launcher launcher) throws IOException, InterruptedException {
+            // TODO could perhaps be more efficient for large files to send a MasterToSlaveFileCallable<byte[]>
+            return IOUtils.toByteArray(getOutputFile(workspace).read());
```

Comment:
Do IOUtils have max heap size protection here or is it possible to allocate to much?